typed_sql/lib/src/typed_sql.mutation.dart

```diff
@@ -0,0 +1,299 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+part of 'typed_sql.dart';
+
+final class UpdateSet<T extends Model> {
+  final List<Expr?> _values;
+
+  UpdateSet._(this._values);
+}
+
+/// A `UPDATE` statement to update at-most one row.
+final class UpdateSingle<T extends Model> {
+  final Update<T> _update;
+
+  UpdateSingle._(this._update);
+
+  /// Execute this `UPDATE` statement in the database.
+  Future<void> execute() async => await _update.execute();
+
+  /// Create a `UPDATE` statement that returns a projection of the update row,
+  /// using the `RETURNING` clause.
+  ///
+  /// This is equivalent to the `RETURNING` clause in SQL.
+  ///
+  /// The [projectionBuilder] must return a [Record] where values are [Expr]
+  /// objects.
+  ReturnSingle<S> returning<S extends Record>(
+    S Function(Expr<T> updated) projectionBuilder,
+  ) =>
+      _update.returning(projectionBuilder)._first;
+
+  /// Create a `UPDATE` statement that returns the updated row, using the
+  /// `RETURNING` clause.
+  ///
+  /// This is equivalent to the `RETURNING *` clause in SQL.
+  ReturnSingle<(Expr<T>,)> returnUpdated() => _update.returnUpdated()._first;
+}
+
+/// A `UPDATE` statement to update one or more rows.
```

Comment:
Is this really 
```suggestion
/// A `UPDATE` statement to update multiple rows.
```
(I guess it can also update zero rows)